### PR TITLE
Add go1.18 to ci pipeline

### PR DESCRIPTION
**Pull request by mfridman**
Checking pipeline with upcoming major Go version (1.18rc1)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go: [1.15, 1.16, 1.17, "1.18.0-rc1"]
+        go: [1.15, 1.16, 1.17, 1.18]
```
Comment (mfridman) [inline .github/workflows/build.yml:28]: Let's drop 1.15 ?

Comment (oxisto) [reply .github/workflows/build.yml:28]: Should we bump the `go.mod` version to Go 1.16 then as well?

Comment (oxisto) [reply .github/workflows/build.yml:28]: Or maybe release a new version first, especially with #175 and #139?

**Comment by mfridman** [comment]
I bumped go.mod to 1.16 minimum, and also updated the GitHub actions to v3 (both checkout and go-setup).

Note that stable was removed, so future tags can be used with semver

https://github.com/actions/setup-go/releases/tag/v3.0.0

**Review by oxisto — APPROVED** [review]

**Pull request merged**